Villager Trades using only second buy slot crashes entire server

### What behaviour is expected:
For me to be able to trade with villagers using only the second trade slot (`buyB` in NBT).

### What behaviour is observed:
A server crash:
java.lang.AssertionError: TRAP
        at net.minecraft.server.v1_16_R2.ItemStack.checkEmpty(ItemStack.java:156) ~[patched_1.16.3.jar:git-Paper-229]
        at net.minecraft.server.v1_16_R2.ItemStack.setCount(ItemStack.java:976) ~[patched_1.16.3.jar:git-Paper-229]
        at net.minecraft.server.v1_16_R2.ItemStack.add(ItemStack.java:980) ~[patched_1.16.3.jar:git-Paper-229]
        at net.minecraft.server.v1_16_R2.ItemStack.subtract(ItemStack.java:984) ~[patched_1.16.3.jar:git-Paper-229]
        at net.minecraft.server.v1_16_R2.MerchantRecipe.b(MerchantRecipe.java:212) ~[patched_1.16.3.jar:git-Paper-229]
        at net.minecraft.server.v1_16_R2.SlotMerchantResult.a(SourceFile:58) ~[patched_1.16.3.jar:git-Paper-229]
        at net.minecraft.server.v1_16_R2.Container.b(Container.java:371) ~[patched_1.16.3.jar:git-Paper-229]
        at net.minecraft.server.v1_16_R2.Container.a(Container.java:183) ~[patched_1.16.3.jar:git-Paper-229]
        at net.minecraft.server.v1_16_R2.PlayerConnection.a(PlayerConnection.java:2445) ~[patched_1.16.3.jar:git-Paper-229]
        at net.minecraft.server.v1_16_R2.PacketPlayInWindowClick.a(SourceFile:32) ~[patched_1.16.3.jar:git-Paper-229]
        at net.minecraft.server.v1_16_R2.PacketPlayInWindowClick.a(SourceFile:10) ~[patched_1.16.3.jar:git-Paper-229]
        at net.minecraft.server.v1_16_R2.PlayerConnectionUtils.lambda$ensureMainThread$1(PlayerConnectionUtils.java:23) ~[patched_1.16.3.jar:git-Paper-229]
        at net.minecraft.server.v1_16_R2.TickTask.run(SourceFile:18) ~[patched_1.16.3.jar:git-Paper-229]
        at net.minecraft.server.v1_16_R2.IAsyncTaskHandler.executeTask(IAsyncTaskHandler.java:136) ~[patched_1.16.3.jar:git-Paper-229]
        at net.minecraft.server.v1_16_R2.IAsyncTaskHandlerReentrant.executeTask(SourceFile:23) ~[patched_1.16.3.jar:git-Paper-229]
        at net.minecraft.server.v1_16_R2.IAsyncTaskHandler.executeNext(IAsyncTaskHandler.java:109) ~[patched_1.16.3.jar:git-Paper-229]
        at net.minecraft.server.v1_16_R2.MinecraftServer.ba(MinecraftServer.java:1135) ~[patched_1.16.3.jar:git-Paper-229]
        at net.minecraft.server.v1_16_R2.MinecraftServer.executeNext(MinecraftServer.java:1128) ~[patched_1.16.3.jar:git-Paper-229]
        at net.minecraft.server.v1_16_R2.IAsyncTaskHandler.awaitTasks(IAsyncTaskHandler.java:119) ~[patched_1.16.3.jar:git-Paper-229]
        at net.minecraft.server.v1_16_R2.MinecraftServer.sleepForTick(MinecraftServer.java:1089) ~[patched_1.16.3.jar:git-Paper-229]
        at net.minecraft.server.v1_16_R2.MinecraftServer.w(MinecraftServer.java:1003) ~[patched_1.16.3.jar:git-Paper-229]
        at net.minecraft.server.v1_16_R2.MinecraftServer.lambda$a$0(MinecraftServer.java:177) ~[patched_1.16.3.jar:git-Paper-229]
        at java.lang.Thread.run(Thread.java:832) [?:?]

### Steps/models to reproduce:
`/summon minecraft:wandering_trader ~ ~ ~ {Offers:{Recipes:[{buyB:{id:"minecraft:stick",Count:1b},sell:{id:"minecraft:stick",Count:1b}}]},NoAI:1b}`
And then try to trade.
The reasoning for only using the second slot is that it's not affected by in-game events. The first slot pricing is influenced by restocks, demand and other gameplay style parameters that shouldn't be influencing adventure-style servers.

### Plugin list:
None

### Paper version:
`git-Paper-240 (MC: 1.16.3) (Implementing API version 1.16.3-R0.1-SNAPSHOT)`

### Side Note:
When this gets fixed, how do I update? Do I have to wait for a major release, can I download it instantly from the website, does it update automatically?

--- a/patches/server/0728-Fix-MerchantOffer-BuyB-Only-AssertionError.patch
+++ b/patches/server/0728-Fix-MerchantOffer-BuyB-Only-AssertionError.patch
@@ -1,0 +1,18 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: thamid-gamer <60953955+thamid-gamer@users.noreply.github.com>
+Date: Fri, 16 Jul 2021 16:00:17 -0400
+Subject: [PATCH] Fix MerchantOffer BuyB-Only AssertionError
+
+
+diff --git a/src/main/java/net/minecraft/world/item/trading/MerchantOffer.java b/src/main/java/net/minecraft/world/item/trading/MerchantOffer.java
+index 70b703b920752e7301e4f19cdc07a1a4ceac5e0e..33660209885a9d56b127ca3926b6c7c60469127e 100644
+--- a/src/main/java/net/minecraft/world/item/trading/MerchantOffer.java
++++ b/src/main/java/net/minecraft/world/item/trading/MerchantOffer.java
+@@ -231,6 +231,7 @@ public class MerchantOffer {
+         if (!this.satisfiedBy(firstBuyStack, secondBuyStack)) {
+             return false;
+         } else {
++            if (!this.getCostA().isEmpty()) // Paper
+             firstBuyStack.shrink(this.getCostA().getCount());
+             if (!this.getCostB().isEmpty()) {
+                 secondBuyStack.shrink(this.getCostB().getCount());

